docs/Updater.md

```diff
@@ -0,0 +1,19 @@
+# Updater
+Updater has the main responsibility of updating the application to a newer version. In doing so,
+it does several things:
+
+- update the application and the components it depends on, i.e. the libraries JAR that it uses
+- maintain backups of the application so that user can use the application even if an update fails
+- maintain dependencies of current version and backup versions and clean up dependencies no longer used
+
+## How Updater updates the application
+<img src="images/How Updater Works.jpg" width="600">
+
+## Maintaining Backups and Dependencies
+A backup of the application will be created before applying update, that is, upon closing the application. Dependencies
+of backup versions are stored so that dependencies still used by backup versions will not be deleted erroneously.
+
+Several backups will be kept to allow user to roll back to older versions if update keeps failing. Once the amount of
+backups exceeds the limit of amount of backups stored (as specified in the code), the oldest backup version will be
+removed. If there are dependencies which are no longer used as the oldest backup version is removed, those dependencies
+will be removed as well so as not to take up disk space unnecessarily.
```

Comment:
1. use standard uml in the sequence diagram.
2. also describe a bit more about what are the names of different jar files, where the backups are stored, how are they named, how to revert to an older version, when does the updater runs, when is the update applied.